daemon/cmd/daemon.go

```diff
@@ -537,6 +538,20 @@ func NewDaemon(ctx context.Context, cancel context.CancelFunc, epMgr *endpointma
 	handleNativeDevices(isKubeProxyReplacementStrict)
 	finishKubeProxyReplacementInit(isKubeProxyReplacementStrict)
 
+	// Cgroup v2 hierarchy root used for attachment is different when running on Kind.
+	runsOnKind := false
+	if option.Config.EnableHostReachableServices {
+		isKind, err := k8s.IsProviderKind(k8s.Client(), nodeTypes.GetName())
```

Comment:
this function should not exist. Whatever logic this function is doing, by retrieving the node from kube-apiserver, should be done in:
a) inside a `if k8s.IsEnabled() {`
b) with the node information retrieved from line 511: `if err := k8s.WaitForNodeInformation(); err != nil {`

Reply:
>  b) with the node information retrieved from line 511

I thought about putting this into `k8s.WaitForNodeInformation`, but then I felt bad that I need to extend the `Node` struct and introducing yet another global var for handling this corner case. Having a dedicated function for it seems to be much cleaner approach to me.

Reply:
We should avoid doing `GETs` during the initialization to kube-apiserver if we can.

Reply:
Out of curiosity, what are the risks with `GET` during the init?

Reply:
adds load to kube-apiserver unnecessary